substrate/frame/scheduler/src/lib.rs

```diff
@@ -1224,30 +1224,37 @@ impl<T: Config> Pallet<T> {
 
 		let mut incomplete_since = now + One::one();
 		let mut when = IncompleteSince::<T>::take().unwrap_or(now);
-		let mut executed = 0;
+		let mut is_first = true; // first task from the first agenda.
 
 		let max_items = T::MaxScheduledPerBlock::get();
 		let mut count_down = max;
 		let service_agenda_base_weight = T::WeightInfo::service_agenda_base(max_items);
 		while count_down > 0 && when <= now && weight.can_consume(service_agenda_base_weight) {
-			if !Self::service_agenda(weight, &mut executed, now, when, u32::MAX) {
+			if !Self::service_agenda(weight, &mut is_first, now, when, u32::MAX) {
 				incomplete_since = incomplete_since.min(when);
 			}
+			is_first = false;
```

Comment:
service agenda is supposed to mutate `is_first`. mutating it here too feels like a duplication but anyway it is ok.

Maybe having `service_agenda` just taking a copy of `is_first` would make the code cleaner.

Reply:
Yes
It means if we service an agenda with no task, then service the agenda for the next block we will start if `is_first = false`. but I think it is good.